src/base/snark0.ml

```diff
@@ -57,26 +57,22 @@ struct
 
   module Data_spec = struct
     type ('r_var, 'r_value, 'k_var, 'k_value) t =
-      | ( :: ) :
+      | Data_spec :
           ( 'var
           , 'value
           , field
           , (unit, field) Checked.Types.Checked.t )
           Types.Typ.typ
-          * ('r_var, 'r_value, 'k_var, 'k_value) t
-          -> ('r_var, 'r_value, 'var -> 'k_var, 'value -> 'k_value) t
-      | [] : ('r_var, 'r_value, 'r_var, 'r_value) t
+          -> ('r_var, 'r_value, 'var -> 'r_var, 'value -> 'r_value) t
```

Comment:
at this point why not just track it as `('r_var, 'r_value, '_var, '_value) t`?

Reply:
(I think my comment is moot, I glanced and you seem to remove that altogether later)